script.js

```diff
@@ -13,4 +13,44 @@ for (let i = 0; i < links.length; i += 1) {
   links[i].addEventListener('click', () => {
     toolbar.classList.add('display-mobile');
   });
-}
\ No newline at end of file
+}
+
+// Email validation
+const form = document.getElementById('contact-form');
+const errorMsg = document.getElementById('text');
+const email = document.getElementById('email');
+
+form.addEventListener('submit', (event) => {
+  const pattern = /^[a-z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-z0-9-]+(?:\.[a-z0-9-]+)*$/;
+  if (email.value.match(pattern)) {
+    errorMsg.innerHTML = 'This is a valid Email';
+    errorMsg.style.color = '#00ff00';
+  } else {
+    event.preventDefault();
+    errorMsg.innerHTML = 'Error Email Not Valid: You must use lowercase in the email field';
+    errorMsg.style.color = '#ff0000';
+  }
+});
+
+// preserve data in the browser
+const inputName = document.getElementById('name');
+const inputEmail = document.getElementById('email');
+const inputMsg = document.getElementById('msg');
```

Comment:
The form data for all inputs should be stored in a single object, saved to a single local storage registry.
For example:
```
const formData = {
    name: nameInput.value,
    email: emailInput.value,
    body: bodyInput.value,
};
```

Reply:
Thanks for the feedback , For the above review , I did that same way,could you please explain me in details 
const data = {
    name: inputName.value,
    email: inputEmail.value,
    message: inputMsg.value,
  };